Gems/Terrain/Assets/Shaders/Terrain/TerrainCommon.azsli

```diff
@@ -104,9 +113,24 @@ void CalculateTerrainPosition(
     }
 
     float heightScale = worldData.m_max.z - worldData.m_min.z;
-    float height = worldData.m_min.z + input.m_zposition * heightScale;
+    float height = worldData.m_min.z + input.m_zPosition * heightScale;
+
     worldPosition = float3(GetWorldXYPosition(patchData, input.m_xyPosition), height);
     cullDistance = TerrainSrg::CalculateCullDistance(patchData.m_lodLevel, worldPosition);
+    normal = float3(input.m_xyNormal, 0.0);
+
+    if (o_useTerrainClod)
+    {
+        const float quadSize = patchData.m_xyScale * RcpGridSize;
+        const float quadDiagonal = quadSize * sqrt(2.0);
```

Comment:
There is still sqrt(2.0) can be made into a constant or use a literal with a comment.

Reply:
The compiler will definitely pre-compute sqrt(2.0), I don't think it deserves its own constant.